rules/library.bzl

```diff
@@ -219,6 +220,82 @@ def _canonicalize_swift_version(swift_version):
 
     return swift_version
 
+def _xcframework_build_type(*, linkage, packaging):
+    return (linkage, packaging)
+
+def _xcframework_slice(*, xcframework_name, identifier, platform, platform_variant, supported_archs, build_type, path):
+    linkage, packaging = _xcframework_build_type(**build_type)
+    import_name = "{}-{}".format(xcframework_name, identifier)
+    if (linkage, packaging) == ("dynamic", "framework"):
+        apple_dynamic_framework_import(
+            name = import_name,
+            framework_imports = native.glob([path + "/**/*"]),
+            deps = [],
+            tags = _MANUAL,
+        )
+    elif (linkage, packaging) == ("static", "framework"):
+        apple_static_framework_import(
+            name = import_name,
+            framework_imports = native.glob([path + "/**/*"]),
+            deps = [],
+            tags = _MANUAL,
+        )
+    elif (linkage, packaging) == ("static", "library"):
+        objc_import(
+            name = import_name,
+            archives = [path],
+            tags = _MANUAL,
+        )
+    else:
+        fail("Unsupported xcframework slice type {} ({}) in {}".format(
+            build_type,
+            identifier,
+            xcframework_name,
+        ))
+    return (platform, platform_variant, supported_archs, import_name)
+
+def _xcframework(*, library_name, name, slices):
+    xcframework_name = "{}-import-{}.xcframework".format(library_name, name)
```

Comment:
So we add a dep on xcframework per slice? Do apple not have a notion of fat multi platform xcframeworks - or is this uncommon?

Reply:
they can be fat -- they can support multiple archs